internal/packages/status/status.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package status
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/packages/changelog"
+	"github.com/elastic/elastic-package/internal/registry"
+)
+
+// PackageStatus holds version and deployment information about a package
+type PackageStatus struct {
+	Name       string
+	Changelog  []changelog.Revision
+	Local      *packages.PackageManifest
+	Production []packages.PackageManifest
+	Staging    []packages.PackageManifest
+	Snapshot   []packages.PackageManifest
+}
+
+// LocalPackage returns the status of a given package including local development information
+func LocalPackage(packageRootPath string, showAll bool) (*PackageStatus, error) {
+	manifest, err := packages.ReadPackageManifestFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package manifest failed")
+	}
+	changelog, err := changelog.ReadChangelogFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package changelog failed")
+	}
+	status, err := Package(manifest.Name, showAll)
+	if err != nil {
+		return nil, err
+	}
+	status.Changelog = changelog
+	status.Local = manifest
+	return status, nil
+}
+
+// Package returns the status of a given package
+func Package(packageName string, showAll bool) (*PackageStatus, error) {
```

Comment:
nit: this function can be renamed to `RemotePackage`, as it fetches data from the Package Registry